basic-services/ReadPropertyResult.cs

```diff
@@ -39,42 +40,130 @@ public struct ReadPropertyResult
         // Helper 
         public bool IsReliable => Reliability == Reliable;
 
-        internal ReadPropertyResult(JToken token, CultureInfo cultureInfo, string reliability = null, string priority = null)
+        private CultureInfo _CultureInfo;
+
+        internal ReadPropertyResult(Guid id, JToken token, string attribute, CultureInfo cultureInfo = null, string reliability = null, string priority = null)
         {
+            Id = id;
+            Attribute = attribute;
             Reliability = reliability ?? Reliable;
             Priority = priority;
-            
+            StringValue = null;
+            NumericValue = 1;
+            ArrayValue = null;
+            BooleanValue = false;
+            _CultureInfo = cultureInfo;
+
+            ReadToken(token);
+        }
+
+        /// <summary>Parses the JToken type and assigns struct values appropriately.</summary>
+        internal void ReadToken(JToken token)
+        {
+            if (token == null) {
+                NumericValue = 1;
+                StringValue = "Unsupported Data Type";
+                return;
+            }
             // switch on token type and set the fields appropriately
             switch (token.Type)
             {
                 case JTokenType.Integer:
                     NumericValue = token.Value<double>();
-                    StringValue = NumericValue.ToString(cultureInfo);
+                    StringValue = NumericValue.ToString(_CultureInfo);
                     BooleanValue = Convert.ToBoolean(NumericValue);
                     ArrayValue = null;
-
                     break;
                 case JTokenType.Float:
-                    //todo
+                    NumericValue = token.Value<double>();
+                    StringValue = NumericValue.ToString(_CultureInfo);
+                    BooleanValue = Convert.ToBoolean(NumericValue);
+                    break;
                 case JTokenType.String:
-                    // todo
+                    NumericValue = 0;
+                    StringValue = token.Value<string>();
+                    break;
                 case JTokenType.Array:
-                    // todo
+                    ReadArray(token);
+                    break;
                 case JTokenType.Boolean:
-                    // todo
+                    if ((bool)(token) == true) {
+                        NumericValue = 1;
+                        StringValue = "True";
```

Comment:
I'd write this as 

```suggestion
                        BooleanValue = true;
                        StringValue = Convert.ToString(BooleanValue);
```

Similar suggestion for false case.